lab5/commands/FilterByGovernment.java

```diff
@@ -0,0 +1,56 @@
+package org.example.commands;
+
+import org.example.collections.City;
+import org.example.collections.Government;
+import org.example.utils.CollectionManager;
+
+import java.util.Scanner;
+
+public class FilterByGovernment implements Commands {
+
+    private final CollectionManager collectionManager;
+
+    public FilterByGovernment(CollectionManager collectionManager) {
+        this.collectionManager = collectionManager;
+    }
+
+    
+    @Override
+    public void execute() {
+        System.out.println("Please provide the government type to filter by (e.g., DEPOTISM, NOOCRACY, TIMOCRACY): ");
+        Scanner scanner = new Scanner(System.in);
+        String governmentInput = scanner.nextLine().toUpperCase().trim();
+
+        try{
+            Government government = Government.valueOf(governmentInput);
+            collectionManager.getCities().stream()
+                    .filter(city -> city.getGovernment() != null && city.getGovernment() == government)
+                    .forEach(city -> System.out.println());
```

Comment:
it's fine to handle both cases but seperate the common part into a seperate method